common/telemetry/grpc.go

```diff
@@ -166,10 +148,11 @@ func (c *customServerStatsHandler) HandleRPC(ctx context.Context, stat stats.RPC
 			span.SetAttributes(attribute.Key("rpc.request.type").String(msgType))
 		}
 	case *stats.OutPayload:
+		span := trace.SpanFromContext(ctx)
+		c.annotateTags(ctx, span, s.Payload)
```

Comment:
This is the main change: extracting the tags from the gRPC _response_.

Reply:
And _could_ happen that it overrides a tag set by the request, I think that's alright for now.